It table exists but the fixture was not setup, we should reset the table

HavokInspiration opened the pull request:
If you are running your tests and they stop because of an error, the database keeps the state it had prior to the error, which can lead to more problems when you start the test suite again.
This change will force the tables already in the database to be dropped and created again when you restart the test suite.

Refs #8187

HavokInspiration committed:
It table exists but the fixture was not setup, we should reset the table

half2me commented:
yay! :)

markstory commented on src/TestSuite/Fixture/FixtureManager.php:
```diff
@@ -227,14 +227,15 @@ protected function _loadFixtures($test)
         $exists = in_array($table, $sources);
 
-        if ($drop && $exists) {
+        if (($drop && $exists) || ($exists && !$isFixtureSetup)) {
```
Should you also check $drop on the new arm? If the test has turned off dropTables we don't want to drop the tables even if the fixture isn't setup as its likely we don't have enough schema data to rebuild the table.

HavokInspiration replied:
When we hit that ``if`` check, ``isFixtureSetup`` is ``false`` everytime so if we add ``$drop`` to the check, we would end up with the two parts of the ``if`` statement being exactly the same (``$exists && $drop``). (Don't know if I'm being super clear here).

To address your concern, I'd rather we add a new part to check if ``$fixture->schema()`` returns an instance of ``Table`` to see if we can rebuild the table.

What do you think ?

markstory replied:
You're right both sides of the condition would be the same. Checking that the fixture has schema is a better approach.

HavokInspiration committed:
Fixture table should be dropped only if we can rebuild it later

If the fixture `_schema` property is not an instance of Table, it will not be possible to restore the table later on, so the table should not be dropped in that case.

HavokInspiration commented:
@markstory Done. I added an ``instanceof`` check.
